majilove/models.py

```diff
@@ -106,3 +108,402 @@ def display_blood_type(self):
+        }
+
+    @property
+    def leader_skill(self):
+        if self.leader_skill_color is None: return None
+        return self.LEADER_SKILL_INFO['template'].format(**self.leader_skill_variables)
+
+    @property
+    def japanese_leader_skill(self):
+        if self.leader_skill_color is None: return None
+        return self.LEADER_SKILL_INFO['japanese_template'].format(**self.leader_skill_variables)
+
+
+    # Skills
+    SKILL_TYPES = OrderedDict([
+        (1, {
+            'translation': _(u'Score notes'),
+            'english': 'Score notes',
+            'japanese_translation': u'スコアノーツ',
+            'icon': 'scoreup',
+
+            'variables': ['note_count'],
+            'template': _(u'Score Notes +{note_count}'),
+            'japanese_template': u'スコアノーツを{note_count}個追加'
+        }),
+        (2, {
+            'translation': _(u'Perfect score up'),
+            'english': 'Perfect score up',
+            'japanese_translation': u'JUST PERFECTのスコア',
+            'icon': 'scoreup',
+
+            'variables': ['percentage'],
+            'template': _(u'Perfect Score +{percentage}%'),
+            'japanese_template': u'JUST PERFECTのスコア{percentage}%上昇'
+        }),
+        (3, {
+            'translation': _(u'Cut-in'),
+            'english': 'Cut-in',
+            'japanese_translation': u'カットイン',
+            'icon': 'scoreup',
+
+            'variables': ['percentage'],
+            'template': _(u'Cut-in Bonus Score +{percentage}%'),
+            'japanese_template': u'カットインボーナスのスコア{percentage}%上昇'
+        }),
+        (4, {
+            'translation': _(u'Good lock'),
+            'english': 'Good lock',
+            # need someone to check this, feels too long
+            'japanese_translation': u'BADをGREATに',
+            'icon': 'perfectlock',
+
+            'variables': ['note_count'],
+            'template': _(u'Bad > Good ({note_count} Times)'),
+            'japanese_template': _(u'BADを{note_count}回GREATにする')
+        }),
+        (5, {
+            'translation': _(u'Great lock'),
+            'english': 'Great lock',
+            # also seems long, need to check which comma to use
+            'japanese_translation': u'BAD,GREATをPERFECTに',
+            'icon': 'perfectlock',
+
+            'variables': ['note_count'],
+            'template': _(u'Bad/Good > Great ({note_count} Times)'),
+            'japanese_template': u'BAD,GREATを{note_count}回PERFECTにする'
+        }),
+        (6, {
+            'translation': _(u'Healer'),
+            'english': 'Healer',
+            'japanese_translation': u'ライフ回復ノーツ',
+            'icon': 'healer',
+
+            'variables': ['note_count'],
+            'template': _(u'Stamina Recovery Notes +{note_count}'),
+            'japanese_template': u'ライフ回復ノーツを{note_count}個追加'
+        })
+    ])
+
+    ALL_VARIABLES = { item: True for sublist in [ _info['variables'] for _info in SKILL_TYPES.values() ] for item in sublist }.keys()
```

Comment:
Is this something you'll then pass to Javascript to hide/show the fields in the form depending on the selected skill type?

If so I'd just recommend making it explicit like so:

```python
SKILL_VARIABLES = ['percentage', 'note_count', ...]
```